service/github/github.go

```diff
@@ -307,11 +315,43 @@ func (g *PullRequest) setPostedComment(ctx context.Context) error {
 		if c.GetSubjectType() == "line" {
 			line = c.GetLine()
 		}
-		g.postedcs.AddPostedComment(c.GetPath(), line, c.GetBody())
+		if meta := extractMetaComment(c.GetBody()); meta != nil {
+			g.postedcs.AddPostedComment(c.GetPath(), line, meta.GetFingerprint())
+		}
+	}
+	return nil
+}
+
+func extractMetaComment(body string) *metacomment.MetaComment {
+	prefix := "<!-- __reviewdog__:"
+	for _, line := range strings.Split(body, "\n") {
+		if after, found := strings.CutPrefix(line, prefix); found {
+			if metastring, foundSuffix := strings.CutSuffix(after, " -->"); foundSuffix {
+				meta, err := DecodeMetaComment(metastring)
+				if err != nil {
+					log.Printf("failed to decode MetaComment: %v", err)
+					continue
+				}
+				return meta
+			}
+		}
 	}
 	return nil
 }
 
+func DecodeMetaComment(metaBase64 string) (*metacomment.MetaComment, error) {
```

Comment:
**[golint-circleci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>exported function DecodeMetaComment should have comment or be unexported